services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -0,0 +1,147 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import logging
+
+import duckdb
+from libcommon.constants import PROCESSING_STEP_SPLIT_DUCKDB_INDEX_VERSION
+from libcommon.exceptions import (
+    NoIndexableColumnsError,
+    ParquetResponseEmptyError,
+    PreviousStepFormatError,
+    UnsupportedIndexableColumnsError,
+)
+from libcommon.processing_graph import ProcessingStep
+from libcommon.storage import StrPath
+from libcommon.utils import JobInfo
+from libcommon.viewer_utils.index_utils import create_index_dir_split
+
+from worker.config import AppConfig
+from worker.job_runners.split.split_job_runner import SplitJobRunner
+from worker.utils import (
+    CompleteJobResult,
+    IndexRowsResponse,
+    get_previous_step_or_raise,
+)
+
+STRING_FEATURE_DTYPE = "string"
+VALUE_FEATURE_TYPE = "Value"
+DUCKDB_DEFAULT_DB_NAME = "index.db"
+UNSUPPORTED_FEATURES_MAGIC_STRINGS = ["'binary'", "Audio", "Image"]
+CREATE_SEQUENCE_COMMAND = "CREATE OR REPLACE SEQUENCE serial START 1;"
+DROP_INDEX_COMMAND = "PRAGMA drop_fts_index('data');"
+CREATE_INDEX_COMMAND = "PRAGMA create_fts_index('data', '__id', '*');"
+CREATE_TABLE_COMMAND = "CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS __id, * FROM"
+INSTALL_EXTENSION_COMMAND = "INSTALL '{extension}';"
+LOAD_EXTENSION_COMMAND = "LOAD '{extension}';"
+# TODO: What if __id field already exist?
+
+
+def compute_index_rows(dataset: str, config: str, split: str, duckdb_index_directory: StrPath) -> IndexRowsResponse:
+    logging.info(f"get index-rows for dataset={dataset} config={config} split={split}")
+
+    # get the first rows from previous job
+    upstream_response = get_previous_step_or_raise(
```

Comment:
use the /parquet endpoint to get the features? and thus remove the need to depend on split-first-rows-from-streaming and split-first-rows-from-parquet?

Reply:
Done